warp/Network/Wai/Handler/Warp/WithApplication.hs

```diff
@@ -21,13 +23,18 @@ import           Network.Wai.Handler.Warp.Types
 --
 -- @since 3.2.4
 withApplication :: IO Application -> (Port -> IO a) -> IO a
-withApplication mkApp action = do
+withApplication = withApplicationSettings defaultSettings
+
+-- | 'withApplication' with given 'Settings'
```

Comment:
It's worth including a note that any port provided in the `Settings` value would be ignored.

Also, can you add a `@since` comment for version 3.2.7 to this and the function added below, and update the `ChangeLog.md` file?